extensions/data-plane/data-plane-api/src/main/java/org/eclipse/dataspaceconnector/dataplane/api/transfer/DataPlanePublicApiController.java

```diff
@@ -0,0 +1,113 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+package org.eclipse.dataspaceconnector.dataplane.api.transfer;
+
+import jakarta.ws.rs.Consumes;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.Context;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.MultivaluedMap;
+import jakarta.ws.rs.core.Request;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.core.UriInfo;
+import org.eclipse.dataspaceconnector.dataplane.core.util.QueryParamsUtils;
+import org.eclipse.dataspaceconnector.dataplane.spi.manager.DataPlaneManager;
+import org.eclipse.dataspaceconnector.dataplane.spi.pipeline.OutputStreamDataSink;
+import org.eclipse.dataspaceconnector.dataplane.spi.result.TransferResult;
+import org.eclipse.dataspaceconnector.dataplane.spi.security.TokenValidationClient;
+import org.eclipse.dataspaceconnector.spi.iam.ClaimToken;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.eclipse.dataspaceconnector.spi.response.ResponseStatus;
+import org.eclipse.dataspaceconnector.spi.types.TypeManager;
+import org.eclipse.dataspaceconnector.spi.types.domain.DataAddress;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.DataFlowRequest;
+
+import java.io.ByteArrayOutputStream;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static org.eclipse.dataspaceconnector.dataplane.api.common.ResponseFunctions.internalErrors;
+import static org.eclipse.dataspaceconnector.dataplane.api.common.ResponseFunctions.notAuthorizedErrors;
+import static org.eclipse.dataspaceconnector.dataplane.api.common.ResponseFunctions.validationError;
+import static org.eclipse.dataspaceconnector.dataplane.api.common.ResponseFunctions.validationErrors;
+import static org.eclipse.dataspaceconnector.dataplane.core.schema.DataRequestSchema.METHOD;
+import static org.eclipse.dataspaceconnector.dataplane.core.schema.DataRequestSchema.QUERY_PARAMS;
+
+/**
+ * Controller for synchronous data pull APIs of the data plane.
+ */
+@Consumes({MediaType.APPLICATION_JSON})
```

Comment:
Popose to document more API details using the swagger annotations https://github.com/swagger-api/swagger-core/wiki/Swagger-2.X---Annotations
So far we missed to do so but I think it would help a lot, especially a short sentence about who is thought to call an endpoint in which situation.
thx a lot in advance!

Reply:
@paullatzelsperger mentioned he will do this with the API work he is engaged in.